docs/weave/weave-api-spec/02-transaction-signatures.md

```diff
@@ -0,0 +1,59 @@
+---
+id: tx-sign-spec
+title: Transaction signitures 
+sidebar_label: Transaction signatures 
+---
+
+In this section, we will explain how to sign __transactions__.
+
+## Authentication and Authorization
+
+To execute desired actions and state changes on weave based blockchains, every transaction must be validated, authenticated and then authorized to perform the action. In the realm of blockchains this is achieved by using signatures.
+
+Weave uses `ed25519` signing algorithm to achieve this important feature. 
+After following the steps in [Weave transactions](transaction.md), created **Tx** must be signed. But first Weave *conditions* role in signing must be explained.
+
+[//]: # (TODO move readthedocs documentation to this project and give references here)
+
+Conditions format defined as `(extension, type, data)` in general. 
+
+Condition with __ed25519__ public key signature could be represented as `("sigs", "ed25519", <address>)`. 
+
+Here is the basic Go presentation of condition 
+
+`condition := fmt.Sprintf("%s/%s/%X", extension, type, data)` where __condition__ looks something like `cond:sigs/ed25519/636f6e646974696f6e64617461`.
+
+## Signing transactions
+
+After the steps in [Weave transactions](transaction.md) are followed, created **Tx** is ready to be signed to be authenticated and authorized by *Weave/bnsd*. 
+
+Signature input body format is:
+
+| version     | len(chainID)     | chainID          | nonce                 | signBytes                  |
+|--------:    |--------------    |--------------    |-------------------    |------------------------    |
+| 4bytes      | uint8            | ascii string     | int64 (bigendian)     | serialized transaction     |
```

Comment:
Should we explain how to build a serialized transaction?

Reply:
Signing takes place after creating the tx. Also, this is the way I ordered docs. If you take a look at [01-transaciton.md](01-transaction.md) creating a tx is explained. serialized tx is the byte presentation of tx struct you have.

Reply:
I agree. The last docs do a great job on serialization.
This just needs to focus on how to sign a particular binary signature.